### PR TITLE
implement `isMainModule` as magic, fix `not` and expreval

### PR DESCRIPTION
**Pull request by metagn**
Hastur also adds `--isMain` to the arguments for every test, maybe this could be restricted to files starting with `t`.

Also adds disabled behavior that makes `const` definitions evaluate their values.

### CODE REVIEW

File: src/nimony/sem.nim [hunk 1]
```diff
@@ -1386,12 +1386,23 @@ proc maybeInlineMagic(c: var SemContext; res: LoadResult) =
       inc n # skip the SymbolDef
       if n.kind == ParLe:
         # ^ export marker position has a `(`? If so, it is a magic!
-        c.dest[c.dest.len-1] = n.load
+        let exprStart = c.dest.len-1
+        c.dest[exprStart] = n.load
+        let magicKind = n.exprKind
```
Comment (Araq) [inline src/nimony/sem.nim:1391]: Not required.

File: src/nimony/sem.nim [hunk 2]
```diff
@@ -1386,12 +1386,23 @@ proc maybeInlineMagic(c: var SemContext; res: LoadResult) =
...
         inc n
         while true:
           c.dest.add n
           if n.kind == ParRi: break
           inc n
+        if magicKind in {IsMainModuleX}:
```
Comment (Araq) [inline src/nimony/sem.nim:1397]: Not required. Inlining a magic doesn't have to cause its evaluation!

File: src/nimony/expreval.nim [hunk 3]
```diff
@@ -70,9 +70,9 @@ proc eval*(c: var EvalContext, n: var Cursor): Cursor =
   template propagateError(r: Cursor): Cursor =
     let val = r
     if val.kind == ParLe and val.tagId == ErrT:
-      val
-    else:
       return val
+    else:
+      val
```
Comment (Araq) [inline src/nimony/expreval.nim:75]: Teach the evalution engine here about `isMainModule`.

**Comment by metagn** [comment]
Had to disable the change that made `const`s evaluate their values as `tcstring` uses a cstring conversion which is not implemented in `expreval` yet. Will do in another PR.

**Comment by Araq** [comment]
> Or we could just not test for it until we have a proper VM.

Yes, we will revisit this logic when there some kind of VM.

**Pull request merged**